packages/opentelemetry-core/src/platform/node/id.ts

```diff
@@ -14,23 +14,27 @@
  * limitations under the License.
  */
 
-import * as crypto from 'crypto';
-
 const SPAN_ID_BYTES = 8;
 const TRACE_ID_BYTES = 16;
 
 /**
  * Returns a random 16-byte trace ID formatted/encoded as a 32 lowercase hex
  * characters corresponding to 128 bits.
  */
-export function randomTraceId(): string {
-  return crypto.randomBytes(TRACE_ID_BYTES).toString('hex');
-}
+export const randomTraceId = getIdGenerator(TRACE_ID_BYTES);
 
 /**
  * Returns a random 8-byte span ID formatted/encoded as a 16 lowercase hex
  * characters corresponding to 64 bits.
  */
-export function randomSpanId(): string {
-  return crypto.randomBytes(SPAN_ID_BYTES).toString('hex');
+export const randomSpanId = getIdGenerator(SPAN_ID_BYTES);
+
+const SHARED_BUFFER = Buffer.allocUnsafe(TRACE_ID_BYTES);
+function getIdGenerator(bytes: number): () => string {
+  return function generateId() {
+    for (let i = 0; i < bytes / 4; i++) {
+      SHARED_BUFFER.writeUInt32BE((Math.random() * 2 ** 32) >>> 0, i * 4);
```

Comment:
actually the `>>> 0` part is not needed as this happens implicit within `writeUInt32BE`.

Reply:
During a benchmark run I recieved the following error:

```
RangeError [ERR_OUT_OF_RANGE]: The value of "value" is out of range. It must be >= 0 and <= 4294967295. Received 4294967295.2376738
```

I was able to reproduce the error in the repl:

```typescript
> const b = Buffer.alloc(4);
undefined
> b.writeUInt32BE(4294967295.2376738, 0);
Thrown:
RangeError [ERR_OUT_OF_RANGE]: The value of "value" is out of range. It must be >= 0 and <= 4294967295. Received 4294967295.2376738
    at checkInt (internal/buffer.js:58:11)
    at writeU_Int32BE (internal/buffer.js:788:3)
    at Buffer.writeUInt32BE (internal/buffer.js:801:10)
    at repl:1:3
    at Script.runInThisContext (vm.js:116:20)
    at REPLServer.defaultEval (repl.js:405:29)
    at bound (domain.js:419:14)
    at REPLServer.runBound [as eval] (domain.js:432:12)
    at REPLServer.onLine (repl.js:716:10)
    at REPLServer.emit (events.js:228:7) {
  code: 
```

And the following code works as expected

```typescript
> const b = Buffer.alloc(4);
undefined
> b.writeUInt32BE(4294967295.2376738 >>> 0, 0);
4
> b
<Buffer ff ff ff ff>
```

It appears, there is a 1/4294967296 chance of generating a random value outside of the range of the function. We have to keep the `>>> 0` to guarantee the function stays in range.